docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst

```diff
@@ -2472,3 +2472,595 @@ The static type of the key being looked up is thus ``p-2.0.0:MyKey``. The key of
+this definition necessarily extends to sub-values.
+
+In a record value of the form ``Constructor { field1 = v1, ..., fieldn = vn }``,
+we say that ``vi`` is a *trailing None* if for all ``j >= i``, ``vj = None``.
+
+Then upon submission of a command whose target template is defined in an LF 1.17
+package, the validation rules for values are relaxed in the following way:
+
+  - The ``record_id``, ``variant_id`` and ``enum_id`` fields of values, 
+    if present, are only checked against the module and type name of th
+    expected type for that value. The package ID component of these fields is
+    ignored.
+  - In record values where all field names are provided, fields of value None
+    may be omitted.
+  - In record values where not all field names are provided, fields must be
+    provided in the same order as that of the record type definition, and 
+    trailing Nones may be omitted.
+
+These rules apply for all sub-values, even those whose type is defined in an
+LF 1.15 or earlier package.
+
+**Example 1**
+
+Assume a LF 1.17 package called ``example1-1.0.0`` which defines a template called 
+``T`` in a module called ``Main``.
+
+.. code:: daml
+
+    module Main where
+
+    template T
+      with
+        p : Party
+      where
+        signatory p
+
+Assume another package called ``other-1.0.0`` which defines a different template
+also called ``T`` in a module also called ``Main``.
+
+.. code:: daml
+    
+    module Main where
+
+    template T
+      with
+        s : Party
+        i : Int
+      where
+        signatory s
+
+Then the ledger API will accept Create commands for ``example1-1.0.0:Main.T`` whose
+create arguments are annotated with type ``other-1.0.0:T:Main.T``, even though
+the type annotation is wrong. In other words, the following console commands
+succeed:
+
+.. code:: scala
+
+    @ val createCmd = Command(
+        command = Command.Command.Create(
+          value = CreateCommand(
+            templateId = Some(
+              value = Identifier(packageId = packageIdExample1, moduleName = "Main", entityName = "T")),
+            createArguments = Some(
+              value = Record(
+                recordId = Some(
+                  Identifier(
+                    packageId = packageIdOther,
+                    moduleName = "Main",
+                    entityName = "T"
+                  )
+                ),
+                fields = Seq(
+                  RecordField(
+                    label = "p",
+                    value = Some(
+                      value = Value(sum = Value.Sum.Party(value = sandbox.adminParty.toLf))
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+      
+    @ sandbox.ledger_api.commands.submit(Seq(sandbox.adminParty), Seq(createCmd))
+
+This is because the module and type names of the type annotation: ``Main`` and
+``T``, match those of the expected type: ``example1-1.0.0:Main.T``.
```

Comment:
```suggestion
This is because the module and type names of the type annotation, ``Main`` and
``T``, match those of the expected type: ``example1-1.0.0:Main.T``.
```